File: dart/dart-sample/sample/basic/literal.dart
// ignore_for_file: unused_local_variable

void main(List<String> args) {
  // numbers
  var int = 42;
  var float = 3.14;
  print(int);
  print(float);

  var hex = 0XDEADBEEF;
  var exponent = 3.14e6;

  // boolean
  var truth = true && false;
  print(truth);

  // null
  var none;
  print(none);

  // strings
  var name = 'abcd';
  print(name);

  // collections
  var array = [1, 2, 3, 4, 5];
  var record = (42, 'foo', 'bar', 3.1415);
  var dictionary = {'name': 'jack', 'age': 18};
  print(array);
  print(record);
  print(dictionary);
}

File: lua/sample/basic/literal.lua
-- boolean
local truth = true
local lie = false
print(type(truth))

-- null
local none = nil
print(type(none))

-- numbers
-- double precision floating numbers
local integer = 3
local real = 3.1415
print(type(real))

-- strings
local str = "abcd"
print(type(str))

-- functions
local function foo()
  return "foo"
end

print(type(foo))
print(foo())
